submissions/yhorodechna/tiny-js-word/lib.js

```diff
@@ -1,56 +1,122 @@
-const dog = {
-  species: 'dog',
-  name: 'Sonya',
-  gender: 'female',
-  legs: 4,
-  hands: 0,
-  saying: 'Woof!',
+
+function print(message, tag = 'pre') {
+  var element = document.createElement(tag);
+  element.innerHTML = message;
+  console.log('PRINT:', message);
+  document.getElementById('main').appendChild(element);
 }
 
-const cat = {
-  species: 'cat',
-  name: 'Terra',
-  gender: 'Female',
-  legs: 4,
-  hands: 0,
-  saying: 'Meow!',
+function makeGitHubLink(currentUrl, filePath) {
+  var prefix = currentUrl.split('//')[0];
+  var domainComponents = currentUrl.split('//')[1].split('/')[0].split('.');
+  var basePath = currentUrl.split('//')[1].split('/')[1];
+  basePath = '/' + basePath + (basePath.length ? '/' : '');
+  var targetDomain = (domainComponents.length > 1)
+    ? 'github.com/' + domainComponents[0]
+    : domainComponents[0];
+  return prefix + '//' + targetDomain + basePath + (domainComponents.length > 1 ? 'blob/gh-pages/' : '') + filePath;
 }
 
-const woman = {
-  species: 'human',
+(function (elementId) {
+  var element = document.getElementById(elementId);
+  element.href = makeGitHubLink(location.href, 'index.js');
+})('source-code');
+
+
+
+
+function addFriends() {
+  let numberOfFriends = Math.floor(Math.random() * 4);
+  let friends = [];
+  let friend;
+  while (numberOfFriends != 0) {
+    friend = INHABITANTS[Math.floor(Math.random() * INHABITANTS.length)]
+    if (friends.some(element => element === friend) == false) {
+      friends.push(friend)
+    };
+    numberOfFriends--;
+  };
+  if (friends.length > 0) {
+    console.log(friends)
+    return friends.map(friend => `my friend is ${friend.species}, ${friend.gender == 'male' ? 'his' : 'her'} name is ${friend.name}`)
+  } else {
+    return `I have no friends :(`
+  };
+};
```

Comment:
It would read better if was a part of a class definition.

Reply:
I added methods to the base class  related to friends